### PR TITLE
nyancat [recipe]

### PR DESCRIPTION
**Pull request by bach5000**
I tried adding NCSA for the license but HaikuPorter doesn't seem to support it.
But it does support UIUC, and I had to add a license file. Did I handle the licenses correctly?

### CODE REVIEW

File: games-misc/nyancat/nyancat-1.5.2.recipe [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+DESCRIPTION="Nyancat is a CLI application that displays a poptart cat \
+in your terminal. You can simply run the binary standalone or use the \
+telnet server version of the CLI by adding the \"-t\" option."
+HOMEPAGE="https://nyancat.dakko.us"
+COPYRIGHT="2011-2018 Kevin Lange"
+LICENSE="UIUC"
+REVISION="1"
+SOURCE_URI="https://github.com/klange/nyancat/archive/$portVersion.tar.gz"
+CHECKSUM_SHA256="88cdcaa9c7134503dd0364a97fa860da3381a09cb555c3aae9918360827c2032"
+SOURCE_FILENAME="nyancat-$portVersion-$srcGitRev.tar.gz"
```
[the remark this answers is not in the document]
Comment (bach5000) [reply games-misc/nyancat/nyancat-1.5.2.recipe]: Oops. Forgot to remove the srcGitRev. Thanks!

**Comment by scottmc** [comment]
When you come across a license that Haiku doesn't already have, you just create a subdirectory under the directory when you are putting the recipe. 
The put it there, renaming it to be the name of the license.

**Comment by bach5000** [comment]
@fbrosson Would I just add `secondaryArchSuffix` to PROVIDES and REQUIRES and such?

**Comment by bach5000** [comment]
@scottmc So would I put both NCSA and UIUC license? Or are they both the same thing? (because I noticed that the author here http://nyancat.dakko.us/ says he used a "NCSA/UIUC" license

**Comment by fbrosson** [comment]
I've just tested nyancat on x86 primary arch, and it works like a charm. So this means it should also work fine on x86 secondary arch.

File: games-misc/nyancat/nyancat-1.5.2.recipe [hunk 2]
```diff
@@ -0,0 +1,49 @@
+SUMMARY="Your own Nyancat in your terminal"
+DESCRIPTION="Nyancat is a CLI application that displays a poptart cat \
+in your terminal. You can simply run the binary standalone or use the \
+telnet server version of the CLI by adding the \"-t\" option."
+HOMEPAGE="https://nyancat.dakko.us"
```
Comment (fbrosson) [inline games-misc/nyancat/nyancat-1.5.2.recipe]: Suggestion, per [`HOMEPAGE`](https://github.com/haikuports/haikuports/wiki/HaikuPorter-Guidelines#homepage):
```diff
-HOMEPAGE="https://nyancat.dakko.us"
+HOMEPAGE="https://nyancat.dakko.us/"
```

**Comment by fbrosson** [comment]
Regarding the license topic, as well as the variant which would allow to get rid of the `$commandSuffix`  I'll let you do whatever you think is best.

File: games-misc/nyancat/nyancat-1.5.2.recipe [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+BUILD()
+{
+	make
+}
+
+INSTALL()
+{
+	mkdir -p "$commandBinDir"
+	cp src/nyancat "$commandBinDir"
```
Comment (fbrosson) [inline games-misc/nyancat/nyancat-1.5.2.recipe]: Suggestion:
```diff
 INSTALL()
 {
-	mkdir -p "$commandBinDir" 
+	install -m 755 -d "$commandBinDir" "$manDir"/man1
-	cp src/nyancat "$commandBinDir"
+	install -m 755 -t "$commandBinDir" src/nyancat
+	install -m 644 -t "$manDir"/man1 nyancat.1
 }
```

**Comment by bach5000** [comment]
@fbrosson I'll fix these right away once I come back. Thanks for helping :)

**Comment by scottmc** [comment]
For the license, just copy the license text they show in the source code into a file called "NYANCAT".  It sort of looks like BSD 3 CLAUSE, but then diverges.  It's probably a mash up of different licenses, so just easier to keep the text as the author wrote it.  Put NYANCAT in licenses/ off the directory the recipe is in and haikuport will handle it, provided you also has a LICENSE = "NYANCAT" in the recipe.

**Comment by scottmc** [comment]
> 
> 
> For the license, just copy the license text they show in the source code into a file called "NYANCAT". It sort of looks like BSD 3 CLAUSE, but then diverges. It's probably a mash up of different licenses, so just easier to keep the text as the author wrote it. Put NYANCAT in licenses/ off the directory the recipe is in and haikuport will handle it, provided you also has a LICENSE = "NYANCAT" in the recipe.

Nevermind.  Just looked at the UIUC license....  you have it right.

**Comment by alaviss** [comment]
@bach5000 Any progress on this?

**Comment by bach5000** [comment]
@alaviss @fbrosson Sorry for taking so long! Completely forgot about these changes.

**Comment by Begasus** [comment]
Line 45, 46 and 50 have a spcace in front of the line

**Pull request merged**